lib/Twig/Extensions/Node/Trans.php

```diff
@@ -111,7 +127,7 @@ public function compile(Twig_Compiler $compiler)
      *
      * @return array
      */
-    protected function compileString(Twig_NodeInterface $body)
+    protected function compileString(Twig_NodeInterface $body, Twig_Node_Expression_Array $withVars = null, $ignoreStrictCheck = false)
```

Comment:
this typehint is wrong. You code allows any expression, so there is no guarantee that it will be a ``Twig_Node_Expression_Array``